Sources/Rendering/Core/CubeAxesActor/index.js

```diff
@@ -0,0 +1,830 @@
+      let drawit = false;
+      const faceAxis = Math.floor(f / 2);
+      const otherAxis1 = (faceAxis + 1) % 3;
+      const otherAxis2 = (faceAxis + 2) % 3;
+      // only for non degenerate axes
+      if (
+        model.dataBounds[otherAxis1 * 2] !==
+          model.dataBounds[otherAxis1 * 2 + 1] &&
+        model.dataBounds[otherAxis2 * 2] !==
+          model.dataBounds[otherAxis2 * 2 + 1]
+      ) {
+        // for each face transform the center and off center to get a direction vector
+        ptv3[faceAxis] =
+          model.dataBounds[f] - 0.1 * length * faceNormals[f][faceAxis];
+        ptv3[otherAxis1] =
+          0.5 *
+          (model.dataBounds[otherAxis1 * 2] +
+            model.dataBounds[otherAxis1 * 2 + 1]);
+        ptv3[otherAxis2] =
+          0.5 *
+          (model.dataBounds[otherAxis2 * 2] +
+            model.dataBounds[otherAxis2 * 2 + 1]);
+        vec3.transformMat4(tmpv3, ptv3, cmat);
+        ptv3[faceAxis] = model.dataBounds[f];
+        vec3.transformMat4(tmp2v3, ptv3, cmat);
+        vec3.subtract(tmpv3, tmp2v3, tmpv3);
+        vec3.normalize(tmpv3, tmpv3);
+        // tmpv3 now holds the face normal vector
+        drawit = tmpv3[2] > 0.05;
+        // for perspctive we need the view direction to the plane
+        if (!model.camera.getParallelProjection()) {
+          drawit = vec3.dot(tmp2v3, tmpv3) > 0.05;
+        }
+      }
+      if (drawit !== model.lastFacesToDraw[f]) {
+        model.lastFacesToDraw[f] = drawit;
+        changed = true;
+      }
+    }
+    return changed;
+  };
+
+  // update the polydata that represents the boundingd edges and gridlines
+  publicAPI.updatePolyData = (facesToDraw, edgesToDraw, ticks) => {
+    // compute the number of points and lines required
+    let numPts = 0;
+    let numLines = 0;
+    numPts += 8; // always start with the 8 cube points
+
+    // count edgesToDraw
+    let numEdgesToDraw = 0;
+    for (let e = 0; e < 12; e++) {
+      if (edgesToDraw[e] > 0) {
+        numEdgesToDraw++;
+      }
+    }
+    numLines += numEdgesToDraw;
+
+    // add values for gridlines
+    if (model.gridLines) {
+      for (let f = 0; f < 6; f++) {
+        if (facesToDraw[f]) {
+          numPts +=
+            ticks[faceAxes[f][0]].length * 2 + ticks[faceAxes[f][1]].length * 2;
+          numLines +=
+            ticks[faceAxes[f][0]].length + ticks[faceAxes[f][1]].length;
+        }
+      }
+    }
+
+    // now allocate the memory
+    const points = new Float64Array(numPts * 3);
+    const lines = new Uint32Array(numLines * 3);
+
+    let ptIdx = 0;
+    let lineIdx = 0;
+
+    // add the 8 corner points
+    for (let z = 0; z < 2; z++) {
+      for (let y = 0; y < 2; y++) {
+        for (let x = 0; x < 2; x++) {
+          points[ptIdx * 3] = model.dataBounds[x];
+          points[ptIdx * 3 + 1] = model.dataBounds[2 + y];
+          points[ptIdx * 3 + 2] = model.dataBounds[4 + z];
+          ptIdx++;
+        }
+      }
+    }
```

Comment:
isn't `vtkBoundingBox.getCorners()` doing the same ?

Reply:
I always forget to add the comment before marking resolved. So my comments all disappear. Have to remember, marking resolved doesn't also submit the comment. getCorners is close, it does a z changes fastest as opposed to x. Not sure why. But I tend to use the first axis changes fastest (C array convention) so with i,j,k or x,y,z  i,x will the the fastest changing index. And as kharma would have it, it looks like I wrote the getCorners on BoundingBox.

Reply:
I would then suggest to change `vtkBoundingBox.getCorners()`. Corner order does not seem to impact its existing usage and it will be more consistent (also WRT VTK C++).

Reply:
Seb has an open MR on bounding box, are you suggesting I edit it now and let him merge the changes?

Reply:
Yes.
I'm happy to do the merge. I prefer that than forgetting to cleanup that code later :-)
@jourdain WDYT ?

Reply:
If so I'll probably move some of the constants into there as well since BoundingBox is a reasonable place for box related constants

Reply:
Into a constants file I would assume